src/Native/System.Native/pal_networking.cpp

```diff
@@ -287,15 +299,219 @@ extern "C" int32_t GetHostEntryForName(const uint8_t* address, HostEntry* entry)
     return PAL_EAI_SUCCESS;
 }
 
-extern "C" int32_t GetNextIPAddress(void** addressListHandle, IPAddress* endPoint)
+static int ConvertGetHostErrorPlatformToPal(int error)
 {
-    if (addressListHandle == nullptr || endPoint == nullptr)
+    switch (error)
     {
-        return PAL_EAI_BADARG;
+        case HOST_NOT_FOUND:
+            return PAL_HOST_NOT_FOUND;
+
+        case TRY_AGAIN:
+            return PAL_TRY_AGAIN;
+
+        case NO_RECOVERY:
+            return PAL_NO_RECOVERY;
+
+        case NO_DATA:
+            return PAL_NO_DATA;
+
+        default:
+            assert(false && "Unknown gethostbyname/gethostbyaddr error code");
+            return PAL_HOST_NOT_FOUND;
+    }
+}
+
+static void ConvertHostEntPlatformToPal(HostEntry& hostEntry, hostent& entry)
+{
+    hostEntry = {
+        .CanonicalName = reinterpret_cast<uint8_t*>(entry.h_name),
+        .Aliases = reinterpret_cast<uint8_t**>(entry.h_aliases),
+        .AddressListHandle = reinterpret_cast<void*>(&entry),
+        .IPAddressCount = 0,
+        .HandleType = HOST_ENTRY_HANDLE_HOSTENT
+    };
+
+    for (int i = 0; entry.h_addr_list[i] != nullptr; i++)
+    {
+        hostEntry.IPAddressCount++;
+    }
+}
+
+#if HAVE_GETHOSTBYNAME_R
+static int GetHostByNameHelper(const uint8_t* hostname, hostent** entry)
+{
+    assert(hostname != nullptr);
+    assert(entry != nullptr);
+
+    size_t scratchLen = 512;
+    uint8_t* buffer = reinterpret_cast<uint8_t*>(malloc(sizeof(hostent) + scratchLen));
```

Comment:
Below you're checking for buffer == nullptr and returning PAL_NO_MEM for that case; should we do the same here?